x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.js

```diff
@@ -343,6 +344,51 @@ export class DataRecognizer {
     return results;
   }
 
+  async dataRecognizerJobsExist(moduleId) {
+    try {
+      const results = {};
+
+      // Load the module with the specified ID and check if the jobs
+      // in the module have been created.
+      const module = await this.getModule(moduleId);
+      if (module && module.jobs) {
+        // Add a wildcard at the front of each of the job IDs in the module,
+        // as a prefix may have been supplied when creating the jobs in the module.
+        const jobIds = module.jobs.map(job => `*${job.id}`);
+        const { jobsExist } = jobServiceProvider(this.callWithRequest);
+        const jobInfo = await jobsExist(jobIds);
```

Comment:
just to confirm, this list will contain all jobs that match the wildcarded ids. 
as in, a single id may match multiple jobs (because the module has been run multiple times), each with possibly a different earliest and latest timestamp.
The anomaly explorer will be opened with all of these jobs, so possibly mixing the results of modules which were run separately on different indices.
It could also match a custom made jobs that just happens to match the wildcard id.

Just checking that this is desired behaviour. 
Also, i have no alternative solution.

Reply:
👀

Reply:
Yes, if jobs had been created multiple times for the sample data set, then the Anomaly Explorer would show multiple sets of jobs from each run of the data recognizer module. I think that is fair enough. 

Note that the `query` for the sample data  set modules (see #35138) uses a term query on the index e.g. `"filter": [{ "term": { "_index": "kibana_sample_data_logs" } }]`, so they only match on the index created by the 'Add sample data' wizard.